c/src/ml-api-common.c

```diff
@@ -1647,3 +1647,202 @@ ml_option_get (ml_option_h option, const char *key, void **value)
+
+  check_feature_state (ML_FEATURE);
+
+  if (!info) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'info' is NULL. It should be a valid ml_information_h");
+  }
+
+  _info = _ml_info_create (ML_INFO_MODEL);
+  if (!_info)
+    _ml_error_report_return (ML_ERROR_OUT_OF_MEMORY,
+        "Failed to allocate memory for the info handle. Out of memory?");
+
+  *info = (ml_information_h *) _info;
+  return ML_ERROR_NONE;
+}
+
+/**
+ * @brief Set key-value pair in given information handle.
+ * @note If duplicated key is given, the value is updated with the new one.
+ */
+int
+_ml_information_set (ml_information_h information, const char *key, void *value,
+    ml_data_destroy_cb destroy)
+{
+  check_feature_state (ML_FEATURE);
+
+  if (!information) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'information' is NULL. It should be a valid ml_information_h, which should be created by ml_information_create().");
+  }
+
+  if (!key) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'key' is NULL. It should be a valid const char*");
+  }
+
+  if (!value) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'value' is NULL. It should be a valid void*");
+  }
+
+  return _ml_info_set_value ((ml_info_s *) information, key, value, destroy);
+}
+
+/**
+ * @brief Frees the given handle of a ml_information.
+ */
+int
+ml_information_destroy (ml_information_h information)
+{
+  check_feature_state (ML_FEATURE);
+
+  if (!information) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'information' is NULL. It should be a valid ml_information_h, which should be created by ml_information_create().");
+  }
+
+  _ml_info_destroy ((ml_info_s *) information);
+
+  return ML_ERROR_NONE;
+}
+
+/**
+ * @brief Gets the value corresponding to the given key in ml_information instance.
+ */
+int
+ml_information_get (ml_information_h information, const char *key, void **value)
+{
+  check_feature_state (ML_FEATURE);
+
+  if (!information) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'information' is NULL. It should be a valid ml_information_h, which should be created by ml_information_create().");
+  }
+
+  if (!key) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'key' is NULL. It should be a valid const char*");
+  }
+
+  if (!value) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'value' is NULL. It should be a valid void**");
+  }
+
+  return _ml_info_get_value ((ml_info_s *) information, key, value);
+}
+
+/**
+ * @brief Internal function for destroy ml_info_list_s instance.
+ */
+static void
+_ml_info_list_destroy (ml_info_list_s * list)
+{
+  guint i;
+
+  for (i = 0; i < list->length; ++i) {
+    _ml_info_destroy (list->info[i]);
+  }
+
+  g_free (list->info);
```

Comment:
I wonder if it's not necessary.

Reply:
internally we allocates new memory for the list, `_info_list->info = g_try_new0 (ml_info_s *, n);`

Reply:
Thank you very much for the detailed explanation!